examples/text_labels_and_annotations/date.py

```diff
@@ -36,33 +33,42 @@
 # column.
 data = cbook.get_sample_data('goog.npz', np_load=True)['price_data']
 
-fig, ax = plt.subplots()
-ax.plot('date', 'adj_close', data=data)
+fig, axs = plt.subplots(3, 1, figsize=(6.4, 7), constrained_layout=True)
+# common to all three:
+for ax in axs:
+    ax.plot('date', 'adj_close', data=data)
+    # Round to nearest years.
+    datemin = np.datetime64(data['date'][0], 'Y')
+    datemax = np.datetime64(data['date'][-1], 'Y') + np.timedelta64(1, 'Y')
```

Comment:
```suggestion
```
and remove `set_ylim()` below.

By using `MonthLocator(bymonth=)` instead of `MonthLocator(interval=)`, we get nice positions without the need to hack the borders to full years.